src/pycram/ontology.py

```diff
@@ -0,0 +1,507 @@
+    """
+    onto_filename = None
+    """
+    Namespace of the loaded ontology
+    """
+    onto_namespace = None
+
+    @classmethod
+    def print_ontology_class(cls, onto_class):
+        """
+        Print information (ancestors, super classes, subclasses, properties, etc.) of an ontology class
+        """
+        if onto_class is None:
+            return
+        print("-------------------")
+        print(onto_class, type(onto_class))
+        print('Super classes: ', onto_class.is_a)
+        print('Ancestors: ', onto_class.ancestors())
+        print('Subclasses: ', list(onto_class.subclasses()))
+        print('Properties: ', list(onto_class.get_class_properties()))
+        print("Instances: ", list(onto_class.instances()))
+        print("Direct Instances: ", list(onto_class.direct_instances()))
+        print("Inverse Restrictions: ", list(onto_class.inverse_restrictions()))
+
+    @classmethod
+    def browse_ontologies(cls, condition: Callable, func: Optional[Callable] = None, **kwargs):
+        if cls.onto is None:
+            assert False, "Main ontology has not been loaded!"
+
+        do_func = func is not None
+        if condition is None:
+            if do_func:
+                func(cls.onto, **kwargs)
+                for sub_onto in cls.onto.get_imported_ontologies():
+                    func(sub_onto, **kwargs)
+        elif condition(cls.onto, **kwargs):
+            if do_func: func(cls.onto, **kwargs)
+        else:
+            for sub_onto in cls.onto.get_imported_ontologies():
+                if condition(sub_onto, **kwargs) and do_func:
+                    func(sub_onto, **kwargs)
+                    break
+
+    def __init__(self, onto_filename: str, onto_search_path=f"{Path.home()}/ontologies"):
+        """
+        Create the singleton object of OntologyManager class
+
+        :param onto_filename: full name path of to be loaded ontology file
+        :param onto_search_path: directory path from which a possibly existing ontology is searched. This is appended
+        to `onto_path`, a global variable by owlready2 containing a list of directories for searching local copies of
+        ontologies (similarly to python `sys.path` for modules/packages).
+        """
+        Path(onto_search_path).mkdir(parents=True, exist_ok=True)
+        onto_path.append(onto_search_path)
+        onto_name = Path(onto_filename).stem
+        OntologyManager.onto_filename = onto_filename
+
+        # Create an ontology world with parallelized file parsing enabled
+        OntologyManager.onto_world = World(filename=f"{onto_search_path}/{onto_name}.sqlite3", exclusive=False,
+                                           enable_thread_parallelism=True)
+
+        onto_ = OntologyManager.onto_world.get_ontology(onto_filename).load()
+        if onto_.loaded:
+            print(f'Main Ontology [{onto_.base_iri}]\'s name: {onto_.name} has been loaded')
+            OntologyManager.onto = onto_
```

Comment:
why are you not referencing self inside the init?

Reply:
it became instance attribute now